scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSource.java

```diff
@@ -416,14 +415,16 @@ public SourceMetadata<V> getSourceMetadata() {
    */
   public abstract static class BucketedInput<V> implements Serializable {
     private static final Pattern BUCKET_PATTERN = Pattern.compile("(\\d+)-of-(\\d+)");
-
     protected TupleTag<V> tupleTag;
-    protected Map<ResourceId, KV<String, FileOperations<V>>> inputs;
     protected Predicate<V> predicate;
     protected Keying keying;
     // lazy, internal checks depend on what kind of iteration is requested
     protected transient SourceMetadata<V> sourceMetadata = null; // lazy
 
+    private transient Map<ResourceId, KV<String, FileOperations<V>>> inputs;
+    private final Map<Integer, KV<String, FileOperations>> fileOperationsEncoding;
```

Comment:
We can probably give the type param here
```suggestion
    private final Map<Integer, KV<String, FileOperations<V>>> fileOperationsEncoding;
```